base/multidimensional.jl

```diff
@@ -863,3 +863,193 @@ If `dim` is specified, returns unique regions of the array `itr` along `dim`.
         @nref $N A d->d == dim ? sort!(uniquerows) : (1:size(A, d))
     end
 end
+
+"""
+    C, ia, ib, ic = uniqueind(itr, dim)
+
+A function that operates similiarly to `unique(itr,dim)` but returns multiple
+output arguments having the following properties:
+
+C - the unique elements of the array `itr` along the selected dimension `dim`
+ia - a Vector{Int} of indices such that:
```

Comment:
Also, I wonder whether the names of the arguments couldn't be improved.Does `ia` stand for `indices in A` and `ic` for `indices in C`? In that case, as in MATLAB, you'd better rename `itr` to `a`, and explicit this meaning in parentheses.

It also means `ib` doesn't stand for anything. Actually, I think you'd better not return it at all, as it can trivially be computed from `ic` IIUC. At the very least, it should have a more explicit name and be returned as the last result, as it is quite different from `ia` and `ib`.